src/services/youtube-api.ts

```
@@ -65,74 +100,67 @@ export default class {
       throw new Error('No video found.');
     }
 
-    return this.getVideo(firstVideo.id, shouldSplitChapters);
+    return this.getVideo(firstVideo.url, shouldSplitChapters);
   }
 
   async getVideo(url: string, shouldSplitChapters: boolean): Promise<SongMetadata[]> {
-    const video = await this.cache.wrap(
-      this.youtube.videos.get,
-      cleanUrl(url),
-      {
-        expiresIn: ONE_HOUR_IN_SECONDS,
-      },
-    );
+    const result = await this.getVideosByID([String(getYouTubeID(url))]);
+    const video = result.at(0)!; // TODO undefined handling
```

let's just throw here if it's undefined to catch issues early on